### ISSUE TITLE
showsUserLocation broken in iOS

### ISSUE THREAD

**Issue by rt2zz** [lelandrichardson/react-native-maps#19]
when `showsUserLocation={true}` I get a 'unrecognized selector sent to instance' exception.

Commenting out the following line prevents the exception: https://github.com/lelandrichardson/react-native-maps/blob/master/ios/AirMaps/AIRMapManager.m#L274

Still digging in to troubleshoot which selector is unrecognized & potential fix. Let me know if you have any ideas.

**Comment by lelandrichardson** [comment]
hmm. The method should be implemented: https://github.com/lelandrichardson/react-native-maps/blob/master/ios/AirMaps/AIRMap.m#L181

Can you verify that its the `setRegion:animated` selector that's unrecognized?  I did notice that it seems to not be on the public interface of `AIRMap.h` which could be a problem... (though wouldn't be a runtime exception IIRC)

**Comment by rt2zz** [comment]
hmm, I was incorrect about commenting out that line. Here is my full error:
`2016-01-19 15:50:32.527 MapboxTest[26970:2104066] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[MKUserLocation setMap:]: unrecognized selector sent to instance 0x7faa1cd00c50'`

Perhaps the issue is that there is no annotation present to animate?

**Comment by christopherdro** [comment]
@rt2zz Did you add `NSLocationWhenInUseUsageDescription` to info.plist?

Also, are you testing through the simulator or on a device?

If im not mistaken, it won't work through the simulator.

**Comment by rt2zz** [comment]
@christopherdro ye NSLocationWhenInUseUsageDescription is defined, this is on simulator, but for reference showsUserLocation does work with native MapView in simulator (with spoofed position). I will try tomorrow on device and see if that yields anything insightful.

**Comment by christopherdro** [comment]
Im noticing this extra line when comparing the two implementations.
https://github.com/lelandrichardson/react-native-maps/blob/74f4da35f7e2eada6d85e35aad494afbfba4ef98/ios/AirMaps/AIRMap.m#L177

@rt2zz Maybe try uncommenting that and see if it works?
@lelandrichardson Is that maybe some dead code leftover? I don't see it being utilized anywhere.

**Comment by lelandrichardson** [comment]
@christopherdro the `map.followUserLocation` property is used in `AIRMapManager.m`.  Some of this code was sort of copied over from the RN map in core right now, and i haven't really spent much time double checking to see that it works...

**Comment by rt2zz** [comment]
Ok after digging a *little* deeper:

The issue comes from the [line 173](https://github.com/lelandrichardson/react-native-maps/blob/master/ios/AirMaps/AIRMap.m#L173) of AirMap.m

The code around this line is basically identical to RCTMap.m so I am not sure where the digression is, my best guess is that some property or method is not being called on MKMapView.

I am going to let this rest for now but will update here when and if I make any progress in the future.

**Comment by lelandrichardson** [comment]
As soon as I get some time, I'll try and check this out. We definitely want to make sure this feature is working.

**Comment by christopherdro** [comment]
Can confirm this from the AirMapsExplorer demo.
```
terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[MKUserLocation setMap:]: unrecognized selector sent to instance 0x15d1363f0'```

**Comment by jrichardlai** [comment]
https://github.com/lelandrichardson/react-native-maps/pull/23 this should fix it

**Comment by lelandrichardson** [comment]
This should now be available in 0.1.2
